feat: add C ndarray interface and refactor implementation for `stats/base/snanmeanwd`

0PrashantYadav0 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: na
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: passed
  - task: lint_c_examples status: passed
  - task: lint_c_benchmarks status: passed
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes. report:
  - task: run_javascript_examples status: na
  - task: run_c_examples status: na
  - task: run_cpp_examples status: na
  - task: run_javascript_readme_examples status: na
  - task: run_c_benchmarks status: na
  - task: run_cpp_benchmarks status: na
  - task: run_fortran_benchmarks status: na
  - task: run_javascript_benchmarks status: na
  - task: run_julia_benchmarks status: na
  - task: run_python_benchmarks status: na
  - task: run_r_benchmarks status: na
  - task: run_javascript_tests status: na ---

Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   Updates native addon from C++ to C for `stats/base/snanmeanwd`
-   adds C ndarray interface for `stats/base/snanmeanwd`
-   refactors JavaScript implementation
-   updates tests, docs, examples and benchmarks

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

0PrashantYadav0 committed:
feat: add C ndarray interface and refactor implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
````diff
@@ -180,6 +168,107 @@ console.log( v );
+<!-- C usage documentation. -->
+
+<section class="usage">
+
+### Usage
+
+```c
+#include "stdlib/stats/base/dnanmeanwd.h"
+```
+
+#### stdlib_strided_dnanmeanwd( N, \*X, strideX )
+
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using Welford's algorithm.
+
+```c
+const double x[] = { 1.0, -2.0, 0.0/0.0, 2.0 };
+
+double v = stdlib_strided_dnanmeanwd( 4, x, 1 );
+// returns ~0.33333
````
this is copied from `stats/base/dnanmeanwd` while the PR is based on the package `stats/base/snanmeanwd` please be mindful of errors like these

0PrashantYadav0 replied:
Sure i will make sure nothing like this happen anymore.

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/benchmark/c/benchmark.length.c:
```diff
@@ -94,19 +94,24 @@ static float rand_float( void ) {
 
 	for ( i = 0; i < len; i++ ) {
-		x[ i ] = ( rand_float()*20000.0f ) - 10000.0f;
+		if ( rand_float() < 0.2 ) {
+			x[ i ] = 0.0 / 0.0; // NaN
+		} else {
+			x[ i ] = ( rand_float() * 20000.0 ) - 10000.0;
```
```suggestion
		if ( rand_float() < 0.2f ) {
			x[ i ] = 0.0f / 0.0f; // NaN
		} else {
			x[ i ] = ( rand_float() * 20000.0f ) - 10000.0f;
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/benchmark/c/benchmark.length.c:
```diff
@@ -120,6 +125,44 @@ static double benchmark( int iterations, int len ) {
+	int i;
+
+	for ( i = 0; i < len; i++ ) {
+		if ( rand_float() < 0.2 ) {
+			x[ i ] = 0.0 / 0.0; // NaN
+		} else {
+			x[ i ] = ( rand_float() * 20000.0 ) - 10000.0;
```
```suggestion
		if ( rand_float() < 0.2f ) {
			x[ i ] = 0.0f / 0.0f; // NaN
		} else {
			x[ i ] = ( rand_float() * 20000.0f ) - 10000.0f;
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/examples/c/example.c:
```diff
@@ -17,21 +17,20 @@
 int main( void ) {
 	// Create a strided array:
-	float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
+	const float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
```
```suggestion
	const float x[] = { 1.0f, 2.0f, 0.0f/0.0f, 3.0f, 0.0f/0.0f, 4.0f, 5.0f, 6.0f, 0.0f/0.0f, 7.0f, 8.0f, 0.0f/0.0f };
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/examples/index.js:
```diff
@@ -18,22 +18,19 @@
+var uniform = require( '@stdlib/random/base/uniform' );
+var filledarrayBy = require( '@stdlib/array/filled-by' );
+var bernoulli = require( '@stdlib/random/base/bernoulli' );
+var snanmeanwd = require( '@stdlib/stats/base/snanmeanwd' );
```
```suggestion
var snanmeanwd = require( './../lib' );
```

0PrashantYadav0 committed:
chore: updated according to code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
```diff
@@ -180,6 +168,107 @@ console.log( v );
+const double x[] = { 1.0, -2.0, 0.0f/0.0f, 2.0 };
+
+double v = stdlib_strided_snanmeanwd_ndarray( 4, x, 1, 0 );
+// returns ~0.33333
```
```suggestion
// returns ~0.33333f
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
````diff
@@ -180,6 +168,107 @@ console.log( v );
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using Welford's algorithm and alternative indexing semantics.
+
+```c
+const double x[] = { 1.0, -2.0, 0.0f/0.0f, 2.0 };
````
```suggestion
const float x[] = { 1.0f, -2.0f, 0.0f/0.0f, 2.0f };
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
````diff
@@ -180,6 +168,107 @@ console.log( v );
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using Welford's algorithm.
+
+```c
+const double x[] = { 1.0, -2.0, 0.0f/0.0f, 2.0 };
````
```suggestion
const float x[] = { 1.0f, -2.0f, 0.0f/0.0f, 2.0f };
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
```diff
@@ -180,6 +168,107 @@ console.log( v );
+const double x[] = { 1.0, -2.0, 0.0f/0.0f, 2.0 };
+
+double v = stdlib_strided_snanmeanwd( 4, x, 1 );
+// returns ~0.33333
```
```suggestion
// returns ~0.33333f
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/src/addon.c:
```diff
@@ -0,0 +1,61 @@
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, API_SUFFIX(stdlib_strided_snanmeanwd)( N, X, strideX ), v );
```
```suggestion
	STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_snanmeanwd)( N, X, strideX ), v );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/src/addon.c:
```diff
@@ -0,0 +1,61 @@
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_INT64( env, offsetX, argv, 3 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
+	STDLIB_NAPI_CREATE_DOUBLE( env, API_SUFFIX(stdlib_strided_snanmeanwd_ndarray)( N, X, strideX, offsetX ), v );
```
```suggestion
	STDLIB_NAPI_CREATE_DOUBLE( env, (double)API_SUFFIX(stdlib_strided_snanmeanwd_ndarray)( N, X, strideX, offsetX ), v );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/test/test.ndarray.js:
```diff
@@ -130,8 +126,7 @@ tape( 'the function supports a negative `stride` parameter', function test( t )
 
-	N = floor( x.length / 2 );
-	v = snanmeanwd( N, x, -2, 6 );
+	v = snanmeanwd( 4, x, -2, 6 );
```
```suggestion
	v = snanmeanwd( 5, x, -2, 8 );
```
please don't pass incorrect values into the tests, see how we want to traverse the array via the comments above

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/test/test.ndarray.js:
```diff
@@ -106,15 +104,13 @@ tape( 'the function supports a `stride` parameter', function test( t ) {
 
-	N = floor( x.length / 2 );
-	v = snanmeanwd( N, x, 2, 0 );
+	v = snanmeanwd( 4, x, 2, 0 );
```
```suggestion
	v = snanmeanwd( 5, x, 2, 0 );
```
you should include the `NaN` element in the computation so that we can verify if our implementation is handling it correctly or not

0PrashantYadav0 committed:
chore: updated docs and test input values

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

aayush0325 commented:
can you please make changes here as done in #4994

0PrashantYadav0 committed:
fix: update examples to use correct N values and handle NaN in arrays

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
Done 👍

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/test/test.ndarray.native.js:
```diff
@@ -175,9 +169,8 @@ tape( 'the function supports an `offset` parameter', opts, function test( t ) {
-	N = floor( x.length / 2 );
 
-	v = snanmeanwd( N, x, 2, 1 );
+	v = snanmeanwd( 4, x, 2, 1 );
```
```suggestion
	v = snanmeanwd( 5, x, 2, 1 );
```
see how the array is being traversed by following the commented indexes

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/test/test.snanmeanwd.js:
```diff
@@ -169,9 +163,8 @@ tape( 'the function supports view offsets', function test( t ) {
-	N = floor(x1.length / 2);
 
-	v = snanmeanwd( N, x1, 2 );
+	v = snanmeanwd( 4, x1, 2 );
```
```suggestion
	v = snanmeanwd( 5, x1, 2 );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/test/test.snanmeanwd.js:
```diff
@@ -130,8 +126,7 @@ tape( 'the function supports a negative `stride` parameter', function test( t )
 
-	N = floor( x.length / 2 );
-	v = snanmeanwd( N, x, -2 );
+	v = snanmeanwd( 4, x, -2 );
```
```suggestion
	v = snanmeanwd( 5, x, -2 );
```

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/test/test.snanmeanwd.js:
```diff
@@ -106,15 +104,13 @@ tape( 'the function supports a `stride` parameter', function test( t ) {
 
-	N = floor( x.length / 2 );
-	v = snanmeanwd( N, x, 2 );
+	v = snanmeanwd( 4, x, 2 );
```
```suggestion
	v = snanmeanwd( 5, x, 2 );
```

0PrashantYadav0 committed:
fix: update test cases to use correct input values for snanmeanwd function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/test/test.snanmeanwd.native.js:
```diff
@@ -197,15 +195,13 @@ tape( 'the function supports a `stride` parameter', opts, function test( t ) {
 
-	N = floor( x.length / 2 );
-	v = snanmeanwd( N, x, 2 );
+	v = snanmeanwd( 4, x, 2 );
```
```suggestion
	v = snanmeanwd( 5, x, 2 );
```
see commented indexes for an indication of how the array is being traversed

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/test/test.snanmeanwd.native.js:
```diff
@@ -221,8 +217,7 @@ tape( 'the function supports a negative `stride` parameter', opts, function test
 
-	N = floor( x.length / 2 );
-	v = snanmeanwd( N, x, -2 );
+	v = snanmeanwd( 4, x, -2 );
```
```suggestion
	v = snanmeanwd( 5, x, -2 );
```

aayush0325 reviewed (changes requested):
minor changes needed, otherwise LGTM

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
same comment about the README structure as in #4777. this is very important as it helps us automate a lot of stuff, kindly take care of this as it's easy for me to miss as well :sweat_smile:

0PrashantYadav0 committed:
docs: add C APIs section and related packages information to README

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

0PrashantYadav0 commented:
@aayush0325 Made the changes you asked for please review.

aayush0325 commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
```diff
@@ -180,6 +172,127 @@ console.log( v );
 
 <!-- /.examples -->
 
+<!-- C usage documentation. -->
```
```suggestion
<!-- C interface documentation. -->
```

0PrashantYadav0 committed:
docs: update C usage documentation to C interface documentation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
```diff
@@ -51,84 +51,79 @@ The [arithmetic mean][arithmetic-mean] is defined as
 -   **N**: number of indexed elements.
 -   **x**: input [`Float32Array`][@stdlib/array/float32].
--   **stride**: index increment for `x`.
+-   **strideX**: index increment for `x`.
```
```suggestion
-   **strideX**: stride length for `x`.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
```diff
@@ -180,6 +172,127 @@ console.log( v );
+
+<!-- /.c -->
+
+<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
+
+<section class="related">
+
```
```suggestion
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
````diff
@@ -180,6 +172,127 @@ console.log( v );
+```c
+const float x[] = { 1.0f, -2.0f, 0.0f/0.0f, 2.0f };
+
+double v = stdlib_strided_snanmeanwd( 4, x, 1 );
````
```suggestion
float v = stdlib_strided_snanmeanwd( 4, x, 1 );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
````diff
@@ -180,6 +172,127 @@ console.log( v );
+```c
+const float x[] = { 1.0f, -2.0f, 0.0f/0.0f, 2.0f };
+
+double v = stdlib_strided_snanmeanwd_ndarray( 4, x, 1, 0 );
````
```suggestion
float v = stdlib_strided_snanmeanwd_ndarray( 4, x, 1, 0 );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/benchmark/benchmark.js:
```diff
@@ -21,16 +21,30 @@
+* Returns a random value or `NaN`.
+*
+* @private
+* @returns {number} random number or `NaN`
```
```suggestion
* Returns a random value.
*
* @private
* @returns {number} random number
```

Applies here and elsewhere in this PR.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/examples/c/example.c:
```diff
@@ -17,21 +17,19 @@
 	// Create a strided array:
-	float x[] = { 1.0, 2.0, 0.0/0.0, 3.0, 0.0/0.0, 4.0, 5.0, 6.0, 0.0/0.0, 7.0, 8.0, 0.0/0.0 };
-
+	const float x[] = { 1.0f, 2.0f, 0.0f/0.0f, 3.0f, 0.0f/0.0f, 4.0f, 5.0f, 6.0f, 0.0f/0.0f, 7.0f, 8.0f, 0.0f/0.0f };
```
```suggestion
	const float x[] = { 1.0f, 2.0f, 0.0f/0.0f, 3.0f, 0.0f/0.0f, 4.0f, 5.0f, 6.0f, 0.0f/0.0f, 7.0f, 8.0f, 0.0f/0.0f };

```

0PrashantYadav0 replied:
@kgryte What i need to change in this ?

kgryte replied:
You're missing the blank line before the C comment.

kgryte reviewed (changes requested):
@0PrashantYadav0 Thanks for working on this. Left a few minor comments. Once resolved, we should be able to merge. Thanks, @aayush0325, for the initial review!

0PrashantYadav0 committed:
docs: clarify stride parameter description and update type in examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@kgryte Made the changes you asked for please review and merge. Thank you.

kgryte commented:
@0PrashantYadav0 You need to apply the benchmark file changes to the other benchmarks, as mentioned in the comment above.

0PrashantYadav0 committed:
docs: update rand function documentation to remove NaN reference

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@kgryte Sorry i miss those changes. I updated them in the latest commit.

aayush0325 committed:
chore: add missing newline

aayush0325 committed:
chore: remove mixed tabs

aayush0325 reviewed (dismissed):
lgtm

0PrashantYadav0 committed:
chore: minor changes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

0PrashantYadav0 commented:
@kgryte Made all the changes please review and merge.

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
```diff
@@ -51,84 +51,79 @@ The [arithmetic mean][arithmetic-mean] is defined as
-#### snanmeanwd( N, x, stride )
+#### snanmeanwd( N, x, strideX )
 
 Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using Welford's algorithm.
```
```suggestion
Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using Welford's algorithm.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/snanmeanwd/README.md:
```diff
@@ -180,6 +172,123 @@ console.log( v );
+
+#### stdlib_strided_snanmeanwd( N, \*X, strideX )
+
+Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array `x`, ignoring `NaN` values and using Welford's algorithm.
```
```suggestion
Computes the [arithmetic mean][arithmetic-mean] of a single-precision floating-point strided array, ignoring `NaN` values and using Welford's algorithm.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
